spec/rubocop/cop/lint/to_enum_arguments_spec.rb

```diff
@@ -0,0 +1,125 @@
+      def m(x, y = 1, *args)
+        return to_enum(:m, x, y) unless block_given?
+               ^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when required keyword arg is missing' do
+    expect_offense(<<~RUBY)
+      def m(x, y = 1, *args, required:)
+        return to_enum(:m, x, y, *args) unless block_given?
+               ^^^^^^^^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when optional keyword arg is missing' do
+    expect_offense(<<~RUBY)
+      def m(x, y = 1, *args, required:, optional: true)
+        return to_enum(:m, x, y, *args, required: required) unless block_given?
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when splat keyword arg is missing' do
+    expect_offense(<<~RUBY)
+      def m(x, y = 1, *args, required:, optional: true, **kwargs)
+        return to_enum(:m, x, y, *args, required: required, optional: optional) unless block_given?
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when arguments are swapped' do
+    expect_offense(<<~RUBY)
+      def m(x, y = 1)
+        return to_enum(:m, y, x) unless block_given?
+               ^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when other values are passed for keyword arguments' do
+    expect_offense(<<~RUBY)
+      def m(required:, optional: true)
+        return to_enum(:m, required: something_else, optional: optional) unless block_given?
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when not inside method definition' do
+    expect_no_offenses(<<~RUBY)
+      to_enum(:m)
+    RUBY
+  end
+
+  it 'does not register an offense when method call has a receiver other then `self`' do
+    expect_no_offenses(<<~RUBY)
+      def m(x)
+        return foo.to_enum(:m) unless block_given?
+      end
+    RUBY
+  end
+
+  it 'registers an offense when method is called on `self`' do
+    expect_offense(<<~RUBY)
+      def m(x)
+        return self.to_enum(:m) unless block_given?
+               ^^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
+      end
+    RUBY
+  end
+
+  it 'ignores the block argument' do
+    expect_no_offenses(<<~RUBY)
+      def m(x, &block)
+        return to_enum(:m, x) unless block_given?
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when enumerator is created for other method' do
+    expect_no_offenses(<<~RUBY)
+      def m(x)
+        return to_enum(:not_m) unless block_given?
```

Comment:
I'm not sure what's the use-case for allowing this tbh. If receiver is `self`, then method should match...